samples/adaptive_load/example_session_spec.textproto

```diff
@@ -0,0 +1,109 @@
+# proto-file: adaptive_load/adaptive_load.proto
+# proto-message: AdaptiveLoadSessionSpec
+
+metrics_plugin_configs {
+  name: "example-metrics-plugin"
+  typed_config {
+    [type.googleapis.com/nighthawk.adaptive_load.ExampleMetricsPluginConfig] {
+      address: "x.y.z"
+      credentials: "key-id-123"
+    }
+  }
+}
+
+# For the full list of built-in metric names, see
+# source/adaptive_load/metrics_plugin_impl.cc.
+
+metric_thresholds {
+  metric_spec {
+    metric_name: "latency-ns-mean-plus-2stdev"
+    metrics_plugin_name: "builtin"
+  }
+  threshold_spec {
+    scoring_function {
+      name: "sigmoid"
+      typed_config {
+        [type.googleapis.com/nighthawk.adaptive_load.SigmoidScoringFunctionConfig] {
+          threshold: 50000000.0  # 50 ms
+          k: 0.00000001  # Should be around 1/threshold
+        }
+      }
+    }
+    weight {
+      value: 1.0
+    }
+  }
+}
+
+metric_thresholds {
+  metric_spec {
+    metric_name: "success-rate"
+    metrics_plugin_name: "builtin"
+  }
+  threshold_spec {
+    lower_threshold {
+      value : 0.95
+    }
+    weight {
+      value: 1.0
+    }
+  }
+}
+
+# Informational metrics are recorded in the output but are not taken into account
+# when adjusting the load.
+
+informational_metric_specs {
+  metric_name: "attempted-rps"
+  metrics_plugin_name: "builtin"
+}
+
+informational_metric_specs {
+  metric_name: "achieved-rps"
+  metrics_plugin_name: "builtin"
+}
+
+informational_metric_specs {
+  metric_name: "send-rate"
+  metrics_plugin_name: "builtin"
+}
+
+# Metric defined in example metrics plugin:
+informational_metric_specs {
+  metric_name: "example_metric1"
+  metrics_plugin_name: "example-metrics-plugin"
+}
+
+# CommandLineOptions proto describing the traffic to generate, but without
+# duration, open_loop, or the field to be varied (e.g. requests_per_second),
+# which will be managed by the adaptive load controller.
+nighthawk_traffic_template {
+  uri {
+    value: "[::1]:12345"
+  }
+}
+
+measuring_period {
+  seconds: 4
+}
+
+convergence_deadline {
+  seconds: 180
+}
+
+testing_stage_duration {
+  seconds: 30
+}
+
+step_controller_config {
+  name: "exponential-search"
+  typed_config {
+    [type.googleapis.com/nighthawk.adaptive_load.ExponentialSearchStepControllerConfig] {
+      input_variable_setter {
+        name: "rps"
```

Comment:
I'm not suggesting I think it would be better, just floating a thought:  I wonder if a  templating approach could be convenient here. E.g. the end-user provides something like:

```
{
 "options": {
  ...
  "duration": "10s",
  "rps": @ExponentialSearchStepControllerConfig.computedRps(),
 ....
 }, 
```

Reply:
I see, so you would put a special symbol in any fields of the CommandLineOptions where the controller should vary the value. This would be a very convenient way to express rps and even fault injection headers and the test server response size header. But other cases I had in mind were URI length and header count, and that might open us to needing things like @ExponentialSearchStepControllerConfig.generatedUriString() which then someone would have to implement in C++, or the ability to write an expression like {{'/'+'a'*num}}, or a more complex template mechanism that can replicate a nested proto a certain number of times. The immediate solution to all such problems is C++ code that manipulates the whole CommandLineOption proto, e.g. setting the URI to /aaaaaaaaaa... of length N or inserting N headers.